### PR TITLE
Updated the CS rule about return null; and return;

### LINKED ISSUE #17201: Reconsider the CS rule for returning null in functions
currently, our CS mandate to use `return;` to return `null` from a function.

But there is 2 different cases actually:
- a function without any return value (returning void), for which `return;` makes sense
- a function returning something or `null`, for which `return null;` would be more meaningful

The original discussion said that PHP had no concept of `void` and so that our CS should not distinguish these 2 cases (and so should have a single rule). But PHP 7.1 now has such concept: https://wiki.php.net/rfc/void_return_type

I think we should revisit this rule in our coding standards, to distinguish functions returning void and functions returning something or null.

what do you think ?

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes https://github.com/symfony/symfony/issues/17201

### CODE REVIEW

**Comment by javiereguiluz** [comment]
I picked the `master` branch ... but this should be merged on `2.3`. I'm sorry.

**Comment by xabbuh** [comment]
Can we also add an example that does not return `null`?

**Comment by javiereguiluz** [comment]
@xabbuh I've added the `void` return example.

**Comment by WouterJ** [comment]
👍 

status: reviewed

File: contributing/code/standards.rst [hunk 1]
```diff
@@ -143,8 +143,9 @@ Structure
 * Add a blank line before ``return`` statements, unless the return is alone
   inside a statement-group (like an ``if`` statement);
 
-* Use just ``return;`` instead of ``return null;`` when a function must return
-  void early;
+* Use ``return null;`` when a function explicitly returns ``null`` values and
+  use ``return;`` when the function returns ``void`` values (available from
+  PHP 7.1);
```
Comment (xabbuh) [inline contributing/code/standards.rst]: I am not sure if the last part wouldn't be confusing. Could people think that this isn't usable yet as their code needed to be compatible with older PHP versions too?

Comment (WouterJ) [inline contributing/code/standards.rst]: I indeed think the part between parenthesis can be removed as it doesn't add anything really usefull (and removing it avoids confusion)

**Comment by WouterJ** [comment]
Not related to this PR, but man, we really have to update the example with a more real example. It doesn't make sense at all anymore. 😄

File: contributing/code/standards.rst (hunk 1 above)
Comment (javiereguiluz) [inline contributing/code/standards.rst]: It's been removed. Thanks!

**Comment by xabbuh** [comment]
👍

**Comment by xabbuh** [comment]
Thank you Javier.

**Pull request closed without merging**